PDSim/scroll/core.py

```diff
@@ -1993,11 +1984,83 @@ def lump_energy_balance_callback(self):
         # sign for the lump
         Qnet -= self.HTProcessed.mean_Q
         
+        return [Qnet]        
+
+    def multi_lump_OEB(self):
+        """
+        Defines multi-lump temperatures energy balance 
+        calculations and returns a list of redisuals
+        
+        Notes:
+        Current example considers two lumps
+            Tlumps[0] : Tshell
+            Tlumps[1] : Toil
+        ----
+        """
+        #For the lumps:
+        Qnet = 0.0
+        Qnet_oil = 0.0
+        
+        # Set the heat input to the suction line
+        #self.suction_heating()
+        
+        # HT terms are positive if heat transfer is TO the lump
+        Qnet -= sum([Tube.Q for Tube in self.Tubes])
         
+        # HT Shell to ambient
+        self.Qamb = self.ambient_heat_transfer(self.Tlumps[0])
+        
+        # Heat transfer with the ambient; Qamb is positive if heat is being removed, thus flip the sign
+        Qnet -= self.Qamb
+        
+        # Heat transfer with the gas in the working chambers.  mean_Q is positive
+        # if heat is transfered to the gas in the working chamber, so flip the 
+        # sign for the lump
+        Qnet -= self.HTProcessed.mean_Q
         
+        #HT oil shell
+        self.Rshell_oil = 190 #K/kW  from Chen (2000) - PhD thesis
```

Comment:
This still needs to be done in a more intelligent way...

Reply:
I would set this parameter in the script and retrieve it in this function.

Reply:
Yes, corrected.